components/Layouts/RightBar/index.tsx

```diff
@@ -0,0 +1,42 @@
+import { Button } from '@/components/ui/button';
+import Header from '@/components/ui/header';
+import useGetAllTags from '@/hooks/useGetAllTags';
+import { useRouter } from 'next/router';
+import React, { useCallback } from 'react';
+
+const RightBar = () => {
+  const { data: tags } = useGetAllTags();
+  // const allUniqueTag = tags?.results;
+  const router = useRouter();
+  const { tag } = router.query;
+  console.log(tag);
+
+  const handleClick = useCallback(
+    (e: any, tag: string) => {
+      e.stopPropagation();
+
+      const url = `/tprojects/${tag}`;
+      router.push(url);
+    },
+    [router]
+  );
+  return (
+    <div className="fixed  z-10 h-dvh  items-center px-2 pt-6 md:items-start md:px-8 lg:w-1/4">
+      <Header data="Tags" />
+      {tags?.results.map((item: string, index: number) => {
+        return (
+          <Button
+            key={index}
+            className={`mb-2 mr-2`}
+            variant={item === tag ? 'secondary' : 'default'}
+            onClick={(e) => handleClick(e, item)}
+          >
+            {item}
+          </Button>
```

Comment:
instead of using buttons please use <Badge /> component from shadcnui.

Reply:
also, user should be able to deselect the tag.

Reply:
here are few problems,

1. we are seeing duplicate tags (ex: typescript). you should convert each tag to small letters and choose only the unique tags to display.
2. bottom right corner, we have a "back to top" FAB button. that's not clickable due to the right sidebar.

<img width="1440" alt="Screenshot 2024-03-24 at 8 26 55 PM" src="https://github.com/rohitdasu/projectmate/assets/48400770/63e8f8a4-e982-4b08-8aca-a04833ad2961">

Reply:
a. shadcdui used 
b. deselect function added but need to click doubleClick

1.duplicate issue: 
a. case sensitve issue solved
b.white space issue solved
but tags like (ex: 'next' and 'next.js' and nextjs  ) are seen as unique
2."back to top" FAB button , is clickable now.
@rohitdasu, review it please.

Reply:
@shovonroy173 can you implement deselect on single click? otherwise users might not know the feature does exist